File: Exercicios C/Pilhas/PilhaSimples/PilhaSimples.c
#include <stdio.h>
#include <stdlib.h>

void push(int vet[], int *tam, int tamTotal,int valor){
    if(*tam < tamTotal){
            vet[*tam] = valor;
            (*tam)++;
        }else{
            printf("\nFila Cheia");
        }
}

void pop(int vet[], int *tam){
    if (*tam == 0) {
            printf("\nFila vazia!");
        } else {

            (*tam)--;
        }
}

void mostrar(int vet[], int tam){

    for(int i = tam - 1; i >= 0; i--){
                printf("%d ", vet[i]);
            }
          
            printf("\n");
}

int main(){
    int tam = 0;
    int tamTotal = 7;
    int vet[tamTotal];
    int opc;

    do{ 
        printf("\n----- MENU -----");
        printf("\nDigite 0 para SAIR");
        printf("\nDigite 1 para INSERIR");
        printf("\nDigite 2 para REMOVER");
        printf("\nDigite 3 para MOSTRAR");
        printf("\nDigite uma opcao: ");
        
        int valor;
        scanf("%d", &opc);

        switch (opc) {
            case 1:
                printf("\nDigite o valor a ser inserido: ");
                scanf("%d", &valor);
                push(vet, &tam, tamTotal, valor);
            break;

            case 2:
                pop(vet, &tam);
            break;
            case 3:
                mostrar(vet,tam);
            break;

            case 0:
                printf("SAINDO....");
            break;

            default:
                printf("\nOpcao invalida!");
            }

        } while (opc != 0);
}

File: Exercicios-JAVA/Pilhas/PilhaSimples/PilhaSimples.java
import java.util.Scanner;

public class PilhaSimples {
    private int tam;
    private int tamTotal;
    private int[]vet;

    public PilhaSimples() {
        this.tam = 0;
        this.tamTotal = 7;
        this.vet = new int[tamTotal];
    }

    public void push(int valor){
        if(tam < tamTotal){
            vet[tam] = valor;
            tam++;
        }else{
            System.out.println("Fila Cheia");
        }
    }

    public void pop(){
        if(tam == 0){
            System.out.println("Lista Vazia");
        }else{
            tam--;
        }
    }

    public void mostrar(){
        if (tam == 0) {
            System.out.println("Pilha Vazia!");
        } else {
            System.out.println("Conteúdo da pilha:");
            for (int i = tam - 1; i >= 0; i--) { //Mostra invertido
                System.out.print(vet[i] + " ");
            }
            System.out.println();
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        PilhaSimples pilha = new PilhaSimples();
        int opc;

        do{ 
            System.out.println("----- MENU -----");
            System.out.println("Digite 0 para SAIR");
            System.out.println("Digite 1 para INSERIR");
            System.out.println("Digite 2 para REMOVER");
            System.out.println("Digite 3 para MOSTRAR");
            System.out.print("Digite uma opcao: ");
        
            int valor;
            opc = scanner.nextInt();

            switch (opc) {
                case 1:
                    System.out.print("Digite o valor a ser inserido: ");
                    valor = scanner.nextInt();
                    pilha.push(valor);
                    break;

                case 2:
                    pilha.pop();
                    break;
                case 3:
                    pilha.mostrar();
                    break;

                default:
                    throw new AssertionError();
            }
        } while (opc != 0);
    }
    
}
